lib/validators/date_validator.rb

```diff
@@ -2,7 +2,7 @@
 class DateValidator < ActiveModel::EachValidator
 
   def validate_each(record, attribute, value)
-    if Time.zone.parse(value.to_s).nil?
+    if Time.zone.parse(value.to_s).nil? && !value.nil?
```

Comment:
Unfortunately, this doesn't pass all of the tests, as the value is *automatically* converted to `nil` when there is a date in an invalid format. This was quite surprising to me - but we learn something new every day!

Replace the second check with `!record.date_before_time_cast.blank?` to check whether the date string before trying to convert it was empty or not.